lib/core/presentation/widgets/movie_card.dart

```diff
@@ -7,15 +7,20 @@ class MovieCard extends StatelessWidget {
   Widget build(BuildContext context) {
     return Padding(
       padding: const EdgeInsets.symmetric(vertical: 12.0, horizontal: 4.0),
-      child: Container(
-        height: 180,
-        width: 135,
-        decoration: BoxDecoration(
-          borderRadius: BorderRadius.circular(15),
-          image: const DecorationImage(
-            fit: BoxFit.cover,
-            image: AssetImage(
-              'asset/poster.jpg',
+      child: GestureDetector(
+        onTap: () {
+          Navigator.pushNamed(context, '/details');
```

Comment:
Prefer Navigator.of(context).pushNamed over Navigator.pushNamed(context,....)

The reason is not obvious and you may never encounter any issue with your current approach but when you do, it will suck. `Navigator.of(context)` is a better option in all cases.

You can read more [here](https://stackoverflow.com/a/71616934)